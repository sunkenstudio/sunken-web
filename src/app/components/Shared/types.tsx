export interface StrapiImage {
  typename: string;
  url: string;
}

export interface StrapiStyledImage {
  typename: string;
  media: StrapiImage;
  alt: string;
  borderWidth: string;
  borderColor: string;
  borderRadius: string;
  grayscalePercent: number;
  filterColor: string;
  filterOpacity: number;
}

export interface RichTextParagraph {
  type: string;
  children: RichTextSegment[];
}

type RichTextSegment = {
  type: "text" | "link";
  text?: string;
  url?: string;
  bold?: boolean;
  italic?: boolean;
  children?: RichTextSegment[];
};

export interface StrapiSection {
  typename: string;
  header: string;
  sortOrder: number;
  text: RichTextParagraph[];
  image: StrapiImage;
  caption: string;
  bgImageOpacity: number;
  bgFilterOpacity: number;
  bgImage: StrapiImage;
}

export interface StrapiBrand {
  typename: string;
  primary: string;
  secondary: string;
  accent: string;
  light: string;
  dark: string;
  font: {
    family: string;
    url: string;
  };
}

export interface StrapiFooter {
  typename: string;
  text: RichTextParagraph[];
  facebookUrl: string;
  instagramUrl: string;
  twitterUrl: string;
}

export interface StrapiHero {
  typename: string;
  header: string;
  subheader: string;
  button1: string;
  button2: string;
  text: RichTextParagraph[];
  image: StrapiStyledImage;
  bgImageOpacity: number;
  bgFilterOpacity: number;
  bgImage: StrapiImage;
}

export interface StrapiContact {
  typename: string;
  bgImage: StrapiImage;
  bgImageOpacity: number;
  bgFilterOpacity: number;
}

export interface Client {
  typename: string;
  clientId: string;
  hero: StrapiHero;
  sections: StrapiSection[];
  brand: StrapiBrand;
  footer: StrapiFooter;
  contact: StrapiContact;
}
