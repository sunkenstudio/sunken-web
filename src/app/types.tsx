import { type BlocksContent } from '@strapi/blocks-react-renderer';

export interface StrapiContact {
  typename: string;
  fields: StrapiInputField[];
  sendTo: string;
  header: string;
  bgColor: Color;
  bgImage: StrapiStyledImage;
}

export interface StrapiInputField {
  label: string;
  type:
    | 'text'
    | 'email'
    | 'tel'
    | 'calendar'
    | 'time'
    | 'time_range'
    | 'dropdown'
    | 'money'
    | 'number'
    | 'textarea'
    | 'date';
  options?: string[];
}

export interface StrapiCoordinates {
  lat: number;
  lng: number;
  label?: string;
}

export interface StrapiConfig {
  typename: string;
  isUnderConstruction: boolean;
  isMaintenanceMode: boolean;
}

export interface StrapiImage {
  typename: string;
  url: string;
}

export type Color = 'primary' | 'secondary' | 'accent' | 'light' | 'dark';

export type StrapiBrandColor = {
  color: Color;
} | null;

export interface StrapiStyledImage {
  typename: string;
  media: StrapiImage;
  alt: string;
  border: StrapiBorder;
  filter: StrapiFilter;
}

export type StrapiBorder = {
  width: string | null;
  color: Color;
  radius: string | null;
} | null;

type StrapiFilter = {
  color: Color;
  opacity: number;
  grayscale: number;
} | null;

export interface StrapiStyledButton {
  typename: string;
  type: 'link' | 'modal' | 'pdf' | 'submit';
  text: string;
  href: string;
  icon: string | null;
  color: string;
  bgColor: string;
  border: StrapiBorder;
  shadow: string;
}

export interface StrapiSection {
  typename: string;
  header: string;
  variant: 'left' | 'right' | 'splitLeft' | 'splitRight';
  sortOrder: number;
  text: BlocksContent;
  image: StrapiStyledImage;
  caption: string | null;
  bgImage: StrapiStyledImage | null;
  bgColor: Color;
}

export interface StrapiBrand {
  typename: string;
  primary: string;
  secondary: string;
  accent: string;
  light: string;
  dark: string;
  fonts: {
    typename: string;
    family: string;
    url: string;
  }[];
  companyName: string;
}

export interface StrapiFooter {
  typename: string;
  text: BlocksContent;
  facebookUrl: string;
  instagramUrl: string;
  twitterUrl: string;
}

export interface StrapiHero {
  typename: string;
  header: string;
  variant: 'leftAligned' | 'centerAligned' | 'split';
  subheader: string;
  buttons: StrapiStyledButton[];
  text: BlocksContent;
  image: StrapiStyledImage | null;
  bgImageOpacity: number;
  bgFilterOpacity: number;
  bgImage: StrapiStyledImage;
}

export interface StrapiMapSection {
  typename: string;
  theme: string;
  center: StrapiCoordinates;
  markers: StrapiCoordinates[];
  markerIcon: string;
  zoom: number;
}

export interface Client {
  typename: string;
  clientId: string;
  hero: StrapiHero;
  sections: StrapiSection[];
  brand: StrapiBrand;
  footer: StrapiFooter;
  contact: StrapiContact;
  config: StrapiConfig;
  mapSection: StrapiMapSection;
}
