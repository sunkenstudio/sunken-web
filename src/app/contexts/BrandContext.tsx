import { createContext, useContext, useState } from "react";
import { StrapiBrand } from "../types";

// Define the type for the context value
type BrandContextType = {
  colors: BrandColors;
  fonts: BrandFonts;
  companyName: string | null;
  loadContent: (brandContent: StrapiBrand) => void;
};

const BrandContext = createContext<BrandContextType>({
  colors: {} as BrandColors,
  fonts: {} as BrandFonts,
  companyName: null,
  loadContent: () => {},
});

interface BrandColors {
  primary: string;
  secondary: string;
  accent: string;
  light: string;
  dark: string;
}

interface BrandFonts {
  headers: {
    typename: string;
    family: string;
    url: string;
  };
  body: {
    typename: string;
    family: string;
    url: string;
  };
}

export const BrandProvider = ({ children }: { children: React.ReactNode }) => {
  const [brand, setBrand] = useState<StrapiBrand>({} as StrapiBrand);

  const loadContent = (brandContent: StrapiBrand) => {
    setBrand(brandContent);
  };

  const getColors = (): BrandColors => ({
    primary: brand.primary,
    secondary: brand.secondary,
    accent: brand.accent,
    light: brand.light,
    dark: brand.dark,
  });

  const getFonts = (): BrandFonts => ({
    headers: brand.fonts?.[0],
    body: brand.fonts?.[1] || brand.fonts?.[0],
  });

  return (
    <BrandContext.Provider
      value={{
        colors: getColors(),
        fonts: getFonts(),
        companyName: brand.companyName,
        loadContent,
      }}
    >
      {children}
    </BrandContext.Provider>
  );
};

export const useBrand = () => {
  return useContext(BrandContext);
};
