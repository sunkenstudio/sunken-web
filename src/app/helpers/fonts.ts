import { StrapiBrand } from "../components/Shared/types";

const FontFaceObserver = require("fontfaceobserver");

const Fonts = (font: StrapiBrand["font"]) => {
  const { url, family } = font;
  const link = document.createElement("link");
  link.href = url;
  link.rel = "stylesheet";

  document.head.appendChild(link);

  const roboto = new FontFaceObserver(family);

  roboto.load().then(() => {
    document.documentElement.classList.add(family.toLowerCase());
  });
};

export default Fonts;
