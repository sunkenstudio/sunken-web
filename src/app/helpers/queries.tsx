import { gql } from "@apollo/client";

export const GET_SITE = gql`
  query Site($ClientId: String) {
    sites(filters: { ClientId: { eq: $ClientId } }) {
      data {
        attributes {
          ClientId
          hero {
            data {
              attributes {
                Header
                Subheader
                Button1
                Button2
                Text
                Image {
                  data {
                    attributes {
                      url
                    }
                  }
                }
                BgImageOpacity
                BgFilterOpacity
                BgImage {
                  data {
                    attributes {
                      url
                    }
                  }
                }
              }
            }
          }
          sections {
            data {
              attributes {
                Header
                SortOrder
                Text
                Image {
                  data {
                    attributes {
                      url
                    }
                  }
                }
                Caption
                BgImageOpacity
                BgFilterOpacity
                BgImage {
                  data {
                    attributes {
                      url
                    }
                  }
                }
              }
            }
          }
          brand {
            data {
              attributes {
                Primary
                Secondary
                Accent
                Light
                Dark
                Font {
                  Url
                  Family
                }
              }
            }
          }
          footer {
            data {
              attributes {
                Text
                FacebookUrl
                InstagramUrl
                TwitterUrl
              }
            }
          }
          contact {
            data {
              attributes {
                BgImage {
                  data {
                    attributes {
                      url
                    }
                  }
                }
                BgImageOpacity
                BgFilterOpacity
              }
            }
          }
        }
      }
    }
  }
`;
