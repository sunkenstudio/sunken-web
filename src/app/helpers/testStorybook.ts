// @ts-nocheck
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import preview from '../../../.storybook/preview';

export const testStories = (Stories: any) => {
  const wrapper = preview?.decorators?.[0];
  const testCases = Object.entries(composeStories(Stories));

  test.each(testCases)('Renders %s story', async (_StoryName, Story) => {
    const tree = await render(wrapper(Story, Story));
    expect(tree.baseElement).toMatchSnapshot();
  });
};
