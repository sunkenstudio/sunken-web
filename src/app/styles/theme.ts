import { extendTheme, type ThemeConfig } from "@chakra-ui/react";

const config: ThemeConfig = {
  initialColorMode: "light",
  useSystemColorMode: false,
};

export const breakpoints = {
  sm: "48em",
  md: "62em",
  lg: "80em",
  xl: "96em",
  "2xl": "128em",
};

// 3. extend the theme
const theme = extendTheme({ breakpoints, config });

export default theme;
