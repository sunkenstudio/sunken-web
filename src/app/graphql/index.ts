import { gql } from '@apollo/client';
import * as Queries from './queries';

export const GET_SITE_ID = gql`
  query GetSiteIdByClientId($ClientId: String) {
    sites(filters: { ClientId: { eq: $ClientId } }, pagination: { limit: 1 }) {
      data {
        id
      }
    }
  }
`;
export const GET_SITE = gql`
  ${Queries.GET_HERO}
  ${Queries.SECTIONS}
  ${Queries.PROJECT_SECTION}
  ${Queries.BRAND}
  ${Queries.CAROUSEL}
  ${Queries.FOOTER}
  ${Queries.CONTACT}
  ${Queries.CONFIG}
  ${Queries.IMAGE_SCROLL}
  ${Queries.FEATURE_SECTION}

  query Site($id: ID!) {
    site(id: $id) {
      data {
        attributes {
          ClientId
          hero {
            ...hero
          }
          sections {
            ...sections
          }
          project_section {
            ...projectSection
          }
          brand {
            ...brand
          }
          carousel {
            ...carousel
          }
          footer {
            ...footer
          }
          contact {
            ...contact
          }
          config {
            ...config
          }
          image_scroll {
            ...imageScroll
          }
          feature_section {
            ...featureSection
          }
        }
      }
    }
  }
`;

export const GET_IMAGES = gql`
  query {
    uploadFiles {
      data {
        id
        attributes {
          name
          url
          alternativeText
          caption
        }
      }
    }
  }
`;
