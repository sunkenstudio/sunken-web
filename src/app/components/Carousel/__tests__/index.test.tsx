'use client';
import '@testing-library/jest-dom';
import { render, waitFor } from '@testing-library/react';
import { Carousel, CarouselProps } from '..';
import { StrapiBrand } from '@/app/types';
import { ChakraProvider } from '@chakra-ui/react';
import {
  BrandFixture,
  CarouselFixture,
  ImageFixture,
  StyledImageFixture,
} from '@/app/fixtures';
import { BrandProvider } from '@/app/contexts/BrandContext';
import theme from '@/app/styles/theme';
import { testStories } from '@/app/helpers/testStorybook';
import * as Stories from '../index.stories';

describe('Carousel', () => {
  const renderComponent = (
    overrides: Partial<CarouselProps> = {},
    contextOverrides: Partial<StrapiBrand> = {}
  ) => {
    const defaults = (props: Partial<CarouselProps>): CarouselProps => ({
      carousel: CarouselFixture(),
      ...props,
    });
    return render(
      <ChakraProvider theme={theme}>
        <BrandProvider brandOverrides={BrandFixture(contextOverrides)}>
          <Carousel {...defaults(overrides)} />
        </BrandProvider>
      </ChakraProvider>
    );
  };

  describe('All stories should render successfully', () => {
    testStories(Stories);
  });

  it('matches snapshot', () => {
    const rendered = renderComponent();
    expect(rendered).toMatchSnapshot();
  });

  it('shows arrow buttons if flag is true', () => {
    const rendered = renderComponent();
    const arrowLeft = rendered.queryByLabelText('Show previous image');
    const arrowRight = rendered.queryByLabelText('Show next image');
    expect(arrowLeft).toBeInTheDocument();
    expect(arrowRight).toBeInTheDocument();
  });

  it('does not show arrow buttons if displayArrows is false', () => {
    const rendered = renderComponent({
      carousel: CarouselFixture({
        displayArrows: false,
      }),
    });
    const arrowLeft = rendered.queryByLabelText('Show previous image');
    const arrowRight = rendered.queryByLabelText('Show next image');
    expect(arrowLeft).not.toBeInTheDocument();
    expect(arrowRight).not.toBeInTheDocument();
  });

  it('changes the image if transitionTime is defined', async () => {
    const rendered = renderComponent();
    const firstImage = rendered.getByAltText('Craft cocktail with lemon twist');
    const firstCounter = rendered.getByText('1/3');

    expect(firstImage).toBeInTheDocument();
    expect(firstCounter).toBeInTheDocument();

    await waitFor(
      () => {
        const secondImage = rendered.getByAltText('Outrigger canoe');
        const secondCounter = rendered.getByText('2/3');

        expect(secondImage).toBeInTheDocument();
        expect(secondCounter).toBeInTheDocument();

        const firstImage = rendered.queryByAltText(
          'Craft cocktail with lemon twist'
        );
        expect(firstImage).not.toBeInTheDocument();
      },
      { timeout: 1500 }
    );
  });

  it('does NOT change the image if transitionTime is not defined', async () => {
    const rendered = renderComponent({
      carousel: CarouselFixture({
        transitionTime: undefined,
      }),
    });
    const firstImage = rendered.getByAltText('Craft cocktail with lemon twist');
    const firstCounter = rendered.getByText(`1/3`);

    expect(firstImage).toBeInTheDocument();
    expect(firstCounter).toBeInTheDocument();

    await waitFor(
      () => {
        const secondImage = rendered.queryByAltText('Outrigger canoe');
        expect(secondImage).not.toBeInTheDocument();

        const firstImage = rendered.getByAltText(
          'Craft cocktail with lemon twist'
        );
        const firstCounter = rendered.getByText(`1/3`);

        expect(firstCounter).toBeInTheDocument();
        expect(firstImage).toBeInTheDocument();
      },
      { timeout: 1500 }
    );
  });

  it('increments displayed image if right arrow is clicked', async () => {
    const rendered = renderComponent();
    const arrowRight = rendered.getByLabelText('Show next image');
    const firstImage = rendered.getByAltText('Craft cocktail with lemon twist');
    const firstCounter = rendered.getByText(`1/3`);

    expect(firstImage).toBeInTheDocument();
    expect(firstCounter).toBeInTheDocument();

    arrowRight?.click();
    // wait for transition animation
    await waitFor(
      () => {
        const secondImage = rendered.getByAltText('Outrigger canoe');
        const secondCounter = rendered.getByText('2/3');

        expect(secondImage).toBeInTheDocument();
        expect(secondCounter).toBeInTheDocument();

        const firstImage = rendered.queryByAltText(
          'Craft cocktail with lemon twist'
        );
        expect(firstImage).not.toBeInTheDocument();
      },
      { timeout: 600 }
    );
  });

  it('decrements displayed image if left arrow is clicked', async () => {
    const rendered = renderComponent();
    const arrowLeft = rendered.getByLabelText('Show previous image');
    const firstImage = rendered.getByAltText('Craft cocktail with lemon twist');
    const firstCounter = rendered.getByText(`1/3`);

    expect(firstImage).toBeInTheDocument();
    expect(firstCounter).toBeInTheDocument();

    arrowLeft?.click();
    // wait for transition animation
    await waitFor(
      () => {
        const secondImage = rendered.getByAltText('Dragon boat');
        const secondCounter = rendered.getByText('3/3');

        expect(secondImage).toBeInTheDocument();
        expect(secondCounter).toBeInTheDocument();

        const firstImage = rendered.queryByAltText(
          'Craft cocktail with lemon twist'
        );
        expect(firstImage).not.toBeInTheDocument();
      },
      { timeout: 600 }
    );
  });

  it('hides counter if flag is false', () => {
    const rendered = renderComponent({
      carousel: CarouselFixture({ displayCounter: false }),
    });
    const counter = rendered.queryByText('1/3');
    expect(counter).not.toBeInTheDocument();
  });

  it('renders arrows and does NOT render dots if more than 5 images', () => {
    const rendered = renderComponent({
      carousel: CarouselFixture({
        images: [
          ...CarouselFixture().images,
          StyledImageFixture({
            media: ImageFixture({
              url: 'https://sunkenstudio-strapi-cms.nyc3.digitaloceanspaces.com/8203015ace4f442ef439488dcd11d914.jpeg',
              typename: 'UploadFile',
            }),
            alt: 'drink two',
          }),
          StyledImageFixture({
            media: ImageFixture({
              url: 'https://sunkenstudio-strapi-cms.nyc3.digitaloceanspaces.com/8203015ace4f442ef439488dcd11d914.jpeg',
              typename: 'UploadFile',
            }),
            alt: 'drink three',
          }),
          StyledImageFixture({
            media: ImageFixture({
              url: 'https://sunkenstudio-strapi-cms.nyc3.digitaloceanspaces.com/8203015ace4f442ef439488dcd11d914.jpeg',
              typename: 'UploadFile',
            }),
            alt: 'drink four',
          }),
        ],
      }),
    });
    const firstDotBtn = rendered.queryByLabelText(
      'View Craft cocktail with lemon twist'
    );
    const arrowLeft = rendered.getByLabelText('Show previous image');
    const arrowRight = rendered.getByLabelText('Show next image');

    expect(firstDotBtn).not.toBeInTheDocument();
    expect(arrowLeft).toBeInTheDocument();
    expect(arrowRight).toBeInTheDocument();
  });

  it('changes image when a dot button is clicked', async () => {
    const rendered = renderComponent();
    const firstImage = rendered.getByAltText('Craft cocktail with lemon twist');
    const firstCounter = rendered.getByText(`1/3`);
    const thirdDotBtn = rendered.getByLabelText('view Dragon boat');

    expect(firstImage).toBeInTheDocument();
    expect(firstCounter).toBeInTheDocument();

    thirdDotBtn.click();
    // wait for transition animation
    await waitFor(
      () => {
        const secondImage = rendered.getByAltText('Dragon boat');
        const secondCounter = rendered.getByText('3/3');

        expect(secondImage).toBeInTheDocument();
        expect(secondCounter).toBeInTheDocument();

        const firstImage = rendered.queryByAltText(
          'Craft cocktail with lemon twist'
        );
        expect(firstImage).not.toBeInTheDocument();
      },
      { timeout: 600 }
    );
  });
});
