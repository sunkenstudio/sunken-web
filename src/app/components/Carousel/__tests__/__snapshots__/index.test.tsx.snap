// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel All stories should render successfully Renders Primary story 1`] = `
<body
  class="chakra-ui-light"
>
  <div>
    <div
      class="css-vjwvrt"
      id="carousel"
    >
      <div
        class="outerContainer css-bcub3b"
      >
        <div
          class="innerContainer"
        >
          <div
            class="chakra-offset-slide"
            style="opacity: 0; transform: translateX(300px) translateY(8px) translateZ(0);"
          >
            <div
              class="chakra-aspect-ratio css-1vq9gpo"
            >
              <img
                alt="Craft cocktail with lemon twist"
                class="chakra-image css-g98gbd"
                src="https://sunkenstudio-strapi-cms.nyc3.digitaloceanspaces.com/b181fb8a2e37c5523f5f625b11628738.avif"
              />
            </div>
          </div>
          <span
            class="imageCounter"
          >
            1
            /
            2
          </span>
        </div>
        <div
          class="css-1lads1q"
        >
          <button
            aria-label="Show previous image"
            class="chakra-button css-15bt6gt"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                fill="currentColor"
              />
            </svg>
          </button>
          <button
            aria-label="view Craft cocktail with lemon twist"
            class="chakra-button css-121nryz"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="32"
              viewBox="0 0 256 256"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128,96a32,32,0,1,0,32,32A32,32,0,0,0,128,96Zm0,48a16,16,0,1,1,16-16A16,16,0,0,1,128,144Z"
              />
            </svg>
          </button>
          <button
            aria-label="view Craft cocktail with lemon twist"
            class="chakra-button css-2u9f3v"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="32"
              viewBox="0 0 256 256"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128,96a32,32,0,1,0,32,32A32,32,0,0,0,128,96Zm0,48a16,16,0,1,1,16-16A16,16,0,0,1,128,144Z"
              />
            </svg>
          </button>
          <button
            aria-label="Show next image"
            class="chakra-button css-15bt6gt"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>
  <div
    class="chakra-portal"
  >
    <div
      aria-label="Notifications-top"
      aria-live="polite"
      id="chakra-toast-manager-top"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <div
      aria-label="Notifications-top-left"
      aria-live="polite"
      id="chakra-toast-manager-top-left"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
    <div
      aria-label="Notifications-top-right"
      aria-live="polite"
      id="chakra-toast-manager-top-right"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
    <div
      aria-label="Notifications-bottom-left"
      aria-live="polite"
      id="chakra-toast-manager-bottom-left"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
    <div
      aria-label="Notifications-bottom"
      aria-live="polite"
      id="chakra-toast-manager-bottom"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <div
      aria-label="Notifications-bottom-right"
      aria-live="polite"
      id="chakra-toast-manager-bottom-right"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
  </div>
</body>
`;

exports[`Carousel matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <div
        class="css-vjwvrt"
        id="carousel"
      >
        <div
          class="outerContainer css-bcub3b"
        >
          <div
            class="innerContainer"
          >
            <div
              class="chakra-offset-slide"
              style="opacity: 0; transform: translateX(300px) translateY(8px) translateZ(0);"
            >
              <div
                class="chakra-aspect-ratio css-1vq9gpo"
              >
                <img
                  alt="Craft cocktail with lemon twist"
                  class="chakra-image css-g98gbd"
                  src="https://sunkenstudio-strapi-cms.nyc3.digitaloceanspaces.com/b181fb8a2e37c5523f5f625b11628738.avif"
                />
              </div>
            </div>
            <span
              class="imageCounter"
            >
              1
              /
              2
            </span>
          </div>
          <div
            class="css-1lads1q"
          >
            <button
              aria-label="Show previous image"
              class="chakra-button css-15bt6gt"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              aria-label="view Craft cocktail with lemon twist"
              class="chakra-button css-121nryz"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="32"
                viewBox="0 0 256 256"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128,96a32,32,0,1,0,32,32A32,32,0,0,0,128,96Zm0,48a16,16,0,1,1,16-16A16,16,0,0,1,128,144Z"
                />
              </svg>
            </button>
            <button
              aria-label="view Craft cocktail with lemon twist"
              class="chakra-button css-2u9f3v"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="32"
                viewBox="0 0 256 256"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128,96a32,32,0,1,0,32,32A32,32,0,0,0,128,96Zm0,48a16,16,0,1,1,16-16A16,16,0,0,1,128,144Z"
                />
              </svg>
            </button>
            <button
              aria-label="Show next image"
              class="chakra-button css-15bt6gt"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-label="Notifications-top"
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-top-left"
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-top-right"
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom-left"
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom"
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-bottom-right"
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="css-vjwvrt"
      id="carousel"
    >
      <div
        class="outerContainer css-bcub3b"
      >
        <div
          class="innerContainer"
        >
          <div
            class="chakra-offset-slide"
            style="opacity: 0; transform: translateX(300px) translateY(8px) translateZ(0);"
          >
            <div
              class="chakra-aspect-ratio css-1vq9gpo"
            >
              <img
                alt="Craft cocktail with lemon twist"
                class="chakra-image css-g98gbd"
                src="https://sunkenstudio-strapi-cms.nyc3.digitaloceanspaces.com/b181fb8a2e37c5523f5f625b11628738.avif"
              />
            </div>
          </div>
          <span
            class="imageCounter"
          >
            1
            /
            2
          </span>
        </div>
        <div
          class="css-1lads1q"
        >
          <button
            aria-label="Show previous image"
            class="chakra-button css-15bt6gt"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                fill="currentColor"
              />
            </svg>
          </button>
          <button
            aria-label="view Craft cocktail with lemon twist"
            class="chakra-button css-121nryz"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="32"
              viewBox="0 0 256 256"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128,96a32,32,0,1,0,32,32A32,32,0,0,0,128,96Zm0,48a16,16,0,1,1,16-16A16,16,0,0,1,128,144Z"
              />
            </svg>
          </button>
          <button
            aria-label="view Craft cocktail with lemon twist"
            class="chakra-button css-2u9f3v"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="32"
              viewBox="0 0 256 256"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128,96a32,32,0,1,0,32,32A32,32,0,0,0,128,96Zm0,48a16,16,0,1,1,16-16A16,16,0,0,1,128,144Z"
              />
            </svg>
          </button>
          <button
            aria-label="Show next image"
            class="chakra-button css-15bt6gt"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
